version: '3.8'
services:
  db:
    image: postgres:11-alpine
    platform: linux/amd64
    restart: always
    environment:
      - POSTGRES_USER=twistdbadmin
      - POSTGRES_PASSWORD=76697421
    ports:
      - '5432:5432'
    volumes: 
      - db:/var/lib/postgresql/data
      - ./init-db.sql:/docker-entrypoint-initdb.d/init-db.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U twistdbadmin"]
      interval: 10s
      timeout: 5s
      retries: 5
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"

  redis:
    image: redis
    restart: always
    ports:
      - '6379:6379'
    volumes:
      - redis:/data
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 5s
      retries: 5

  celery:
    build:
      context: .
      dockerfile: Dockerfile.celery.dev
    command: celery -A twist worker -l INFO
    volumes:
      - .:/twist
    depends_on:
      redis:
        condition: service_healthy

  django:
    build:
      context: .
      dockerfile: Dockerfile.dev
    command: sh -c "./dev_setup.py && ./manage.py runserver 0.0.0.0:8000"
    ports:
      - "8000:8000"
    volumes:
      - .:/twist
    depends_on:
      db:
        condition: service_healthy
      redis:
        condition: service_healthy

volumes:
  db:
    driver: local
  redis:
    driver: local

